File: src/2114-Maximum-Number-of-Words-Found-in-Sentences/maximum_number_of_words_found_in_sentences.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>

class Solution {
public:
    int mostWordsFound(std::vector<std::string>& sentences) {
        int ans = 0;

        for (const std::string& s : sentences) {
            int temp = 0;
            std::istringstream iss(s);
            std::string word;

            while (iss >> word) {
                temp += 1;
            }

            if (temp > ans) {
                ans = temp;
            }
        }

        return ans;
    }
};

int main() {
    std::vector<std::string> sentences = {"alice and bob love leetcode", "i think so too", "this is great thanks very much"};
    Solution solution;

    std::cout << solution.mostWordsFound(sentences);

    return 0;
}

File: src/2114-Maximum-Number-of-Words-Found-in-Sentences/maximum_number_of_words_found_in_sentences.py
from typing import List

class Solution:
    def mostWordsFound(self, sentences: List[str]) -> int:
        ans: int = 0

        for s in sentences:
            temp: int = 0
            for w in s:
                if w == ' ':
                    temp += 1
            if temp > ans:
                ans = temp
        
        return ans + 1


if __name__ == "__main__":
    sentences: List[str] = ["alice and bob love leetcode", "i think so too", "this is great thanks very much"]
    solution = Solution()

    print(solution.mostWordsFound(sentences))